src/transforms/mapperTransforms.ts

```diff
@@ -169,18 +185,75 @@ function transformPrimitiveMapper(pipelineValue: PipelineValue) {
   };
 }
 
+// TODO: Make sure this is the correct way to handle additionalProperties
+function getAdditionalProperties(
+  immediateParents: Schema[]
+): Mapper | undefined {
+  return immediateParents.some(p => p.type === SchemaType.Dictionary)
+    ? {
+        type: {
+          name: MapperType.Object
+        }
+      }
+    : undefined;
+}
+
 function transformObjectMapper(pipelineValue: PipelineValue) {
   const { schema, options } = pipelineValue;
 
   if (!isSchemaType([SchemaType.Object], schema)) {
     return pipelineValue;
   }
+  const className = getMapperClassName(schema);
   const objectSchema = schema as ObjectSchema;
-  const modelProperties = processProperties(objectSchema.properties);
+  const { discriminator, discriminatorValue } = objectSchema;
+
+  let modelProperties = processProperties(objectSchema.properties);
+  const parents = objectSchema.parents ? objectSchema.parents.all : [];
+  const immediateParents = objectSchema.parents
+    ? objectSchema.parents.immediate
+    : [];
+  const parentsRefs = immediateParents
+    .map(p => getMapperClassName(p))
+    .filter(p => p !== className);
+
+  const additionalProperties = getAdditionalProperties(immediateParents);
+
+  modelProperties = {
+    ...modelProperties,
+    ...(parentsRefs && parentsRefs.length && { parentsRefs })
+  };
+
+  // This is an uber parent
+  if (discriminator && !objectSchema.parents) {
+    uberParents.push(className);
+  }
+
+  const uberParent = getMapperClassName(
+    parents.find(p => uberParents.includes(getMapperClassName(p))) || schema
+  );
 
   const mapper = buildMapper(
     schema,
-    { name: MapperType.Composite, modelProperties },
+    {
+      name: MapperType.Composite,
+      className,
+      modelProperties,
+      ...(discriminatorValue && {
+        uberParent,
+        polymorphicDiscriminator: `${uberParent}.type.polymorphicDiscriminator`
+      }),
+      ...(discriminator && {
+        // TODO need to handle uber parent in a more appropiate way
```

Comment:
Ignore this comment, will remove it in the next iteration... Now using the uberParents array